.github/workflows/template-creator.yml

```
@@ -0,0 +1,84 @@
+# This workflow should cleanup everything unneeded from the template project
+
+name: Template Creator
+
+on:
+  pull_request:
+  release:
+    types: [published]
+  push:
+    tags:
+    branches:
+      - main
+      - develop
+
+env:
+  TEMPLATES_PATH: ".github/template"
+
+jobs:
+
+  template-create:
+    name: Create new template from existing template
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fetch Sources
+        uses: actions/checkout@v2.4.0
+
+      - name: Get organization and project name 
+        run: |
+          echo "NEW_ORG=${{ github.repository_owner }}" >> $GITHUB_ENV
+          echo "NEW_PROJECT=${{ github.event.repository.name }}" >> $GITHUB_ENV
+          echo "NEW_URL=${{ github.repositoryUrl }}" >> $GITHUB_ENV
+          echo "TEMPLATE_NAME=`cat ${{ env.TEMPLATES_PATH }}/template_name`" >> $GITHUB_ENV
+          echo "TEMPLATE_REPOSITORY=`cat ${{ env.TEMPLATES_PATH }}/template_repository`" >> $GITHUB_ENV
+
+      - uses: octokit/request-action@v2.x
+        id: get_repo_meta
+        with:
+          route: GET /repos/{owner}/{repo}
+          owner: ${{ env.NEW_ORG }}
+          repo: ${{ env.NEW_PROJECT }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # Rename all cpp_starter_project occurrences to current repository and remove this workflow
+      - name: Update repository to match new template information
+        run: |
+          # Update the README and template files to match the new org / repository names 
+          sed -i "s|${{ env.TEMPLATE_REPOSITORY }}|${{ github.repository }}|g" README.md ${{ env.TEMPLATES_PATH }}/template_repository
+          sed -i "s|${{ env.TEMPLATE_NAME }}|${{ github.event.repository.name }}|g" README.md ${{ env.TEMPLATES_PATH }}/template_name
+
+      - name: Print diff after template name replacement
+        run: |
+          git diff
+
+      - name: Setup Cpp
+        uses: aminya/setup-cpp@v1
+        with:
+          compiler: gcc
+
+          cmake: true
+          ninja: false
+          conan: true
+          vcpkg: false
+          ccache: false
+          clangtidy: false
+
+          cppcheck: false
+
+          gcovr: false
+          opencppcoverage: false
+
+      - name: Test simple configuration to make sure nothing broke (default compiler,cmake,developer_mode OFF)
+        run: |
+          cmake -S . -B ./build -DCMAKE_BUILD_TYPE:STRING=DEBUG -DENABLE_DEVELOPER_MODE:BOOL=OFF -DOPT_ENABLE_COVERAGE:BOOL=OFF
+
+      - uses: EndBug/add-and-commit@v4
+        # only commit and push if we are a template and 
+        if: fromJson(steps.get_repo_meta.outputs.data).is_template == true && env.TEMPLATE_REPOSITORY != github.repository
```

I'd use the name as a trigger to commit.
```suggestion
        if: fromJson(steps.get_repo_meta.outputs.data).is_template == true && env.TEMPLATE_NAME != env.NEW_PROJECT
```